ingeniamotion/communication.py

```diff
@@ -708,7 +710,9 @@ def connect_servo_ethercat(
         if not path.isfile(dict_path):
             raise FileNotFoundError(f"Dict file {dict_path} does not exist!")
         if interface_name not in self.mc.net:
-            self.mc.net[interface_name] = EthercatNetwork(interface_name)
+            self.mc.net[interface_name] = EthercatNetwork(
+                interface_name, gil_release_config=gil_release_config
```

Comment:
Passing a reference here to the default dataclass seems like this pitfall:
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

On others methods it is clear that the network is created and destroyed on the same call, but not here.
Maybe you can convert the dataclass to frozen to avoid this problem altogether